specs/phase0/fork-choice.md

```diff
@@ -68,10 +73,34 @@ Any of the above handlers that trigger an unhandled exception (e.g. a failed ass
 ```python
 @dataclass(eq=True, frozen=True)
 class LatestMessage(object):
-    epoch: Epoch
+    slot: Slot
     root: Root
 ```
 
+#### `BlockSlotNode`
+
+```python
+class BlockSlotNode(Container):
+    block_root: Root
+    slot: Slot
+    parent_node: Root
+```
+
+#### `BlockSlotKey`
+
+```python
+class BlockSlotKey(Container):
+    block_root: Root
+    slot: Slot
+```
+
+#### `get_block_slot_key`
+
+```python
+def get_block_slot_key(block_root: Root, slot: Slot) -> Root:
+    return hash_tree_root(BlockSlotKey(block_root=block_root, slot=slot))
```

Comment:
Note to self.  Using `hash_tree_root` here is an implementation detail. Client team is welcome to use something fast and has a good collision resistance

Reply:
Any tree implementation for `block_slot_tree` and appropriately modified helper functions is welcome to use!